npm/test/auth.test.ts

```diff
@@ -312,3 +312,126 @@ test("Throws an error if the access token and refresh token have expired", async
     new LoggedOutError("", "User is logged out")
   )
 })
+
+test("'scheduleTokenRefresh' should refresh tokens if refresh token will expire within the given timeframe", async () => {
+  const updateToken = jest.fn()
+  const isTokenExpired = jest.fn().mockImplementation(() => true)
+  const mockKeycloak: KeycloakInstance = {
```

Comment:
As discussed on the call, it would be good to commonise these mocks if possible.